File: src/main/java/ru/pearx/pmdumper/dumpers/DumperEntities.java
package ru.pearx.pmdumper.dumpers;

import net.minecraft.entity.Entity;
import net.minecraft.entity.EntityList;
import net.minecraft.entity.passive.EntityVillager;
import net.minecraft.util.ResourceLocation;
import net.minecraftforge.fml.common.registry.EntityEntry;
import net.minecraftforge.fml.common.registry.EntityRegistry;
import net.minecraftforge.fml.common.registry.ForgeRegistries;
import net.minecraftforge.fml.common.registry.GameData;
import net.minecraftforge.fml.relauncher.Side;
import net.minecraftforge.fml.relauncher.SideOnly;
import ru.pearx.pmdumper.dumpers.base.IDumper;
import ru.pearx.pmdumper.dumpers.base.PMDData;

import java.util.*;

/**
 * Created by mrAppleXZ on 13.06.17 22:16.
 */
@SideOnly(Side.CLIENT)
public class DumperEntities implements IDumper
{
    @Override
    public String getName()
    {
        return "entities";
    }

    @Override
    public PMDData getData()
    {
        List<List<String>> data = new ArrayList<>();
        Map<String, Integer> counts = new HashMap<>();
        data.add(Arrays.asList("ID", "Name", "Class Name"));
        for(Map.Entry<ResourceLocation, EntityEntry> entr : ForgeRegistries.ENTITIES.getEntries())
        {
            ResourceLocation loc = entr.getKey();
            EntityEntry ent = entr.getValue();
            data.add(Arrays.asList(loc.toString(), ent.getName(), ent.getEntityClass().getName()));
            PMDData.plusCounts(counts, loc.getResourceDomain(), 1);
            EntityVillager.GET_TRADES_DONT_USE()
        }
        return new PMDData(data, counts);
    }
}

File: src/main/kotlin/ru/pearx/pmdumper/dumper/standard/DumperEntities.kt
@file:JvmMultifileClass
@file:JvmName("StandardDumpers")
package ru.pearx.pmdumper.dumper.standard

import net.minecraft.util.ResourceLocation
import net.minecraftforge.fml.common.registry.ForgeRegistries
import ru.pearx.pmdumper.ID
import ru.pearx.pmdumper.dumper.dumper
import ru.pearx.pmdumper.utils.toHexColorString

val DumperEntities = dumper {
    registryName = ResourceLocation(ID, "entities")
    header = listOf("ID", "Name", "Class Name", "Primary Egg Color", "Secondary Egg Color")
    amounts {
        for (entity in ForgeRegistries.ENTITIES)
            this += entity.registryName
    }
    iterator {
        for (entity in ForgeRegistries.ENTITIES) {
            with(ArrayList<String>(header.size)) {
                with(entity) {
                    add(registryName.toString())
                    add(name)
                    add(entityClass.name)
                    add(egg?.primaryColor?.toHexColorString() ?: "")
                    add(egg?.secondaryColor?.toHexColorString() ?: "")
                }
                yield(this)
            }
        }
    }
}